subprojects/execution/src/test/groovy/org/gradle/internal/execution/steps/RemovePreviousOutputsStepTest.groovy

```diff
@@ -189,9 +189,9 @@ class RemovePreviousOutputsStepTest extends StepSpec<InputChangesContext> implem
             visitor.visitOutputProperty("file", TreeType.FILE, outputs.file, TestFiles.fixed(outputs.file))
         }
         _ * context.afterPreviousExecutionState >> Optional.of(afterPreviousExecution)
-        1 * afterPreviousExecution.outputFileProperties >> ImmutableSortedMap.<String, FileCollectionFingerprint>of("dir", outputs.dirFingerprint, "file", outputs.fileFingerprint)
-        1 * outputChangeListener.beforeOutputChange(outputs.dirFingerprint.rootPaths)
-        1 * outputChangeListener.beforeOutputChange(outputs.fileFingerprint.rootPaths)
+        1 * afterPreviousExecution.outputFileProperties >> ImmutableSortedMap.of("dir", outputs.dirSnapshot, "file", outputs.fileSnapshot)
+        1 * outputChangeListener.beforeOutputChange(_ as Iterable) >> { Iterable<String> paths -> paths as List == [outputs.dir.absolutePath] }
```

Comment:
Does this assert anything? It looks like the method returns a boolean and that is it. If you want to have an assertion, I think you need to move the closure into the arguments.

Reply:
Good point, moved.